h/models/moderation_log.py

```diff
@@ -10,17 +12,23 @@
 from h.db.mixins_dataclasses import AutoincrementingIntegerID, CreatedMixin
 from h.models.annotation import ModerationStatus
 
+if TYPE_CHECKING:
+    from h.models.annotation import Annotation
+    from h.models.user import User
+
 
 class ModerationLog(Base, AutoincrementingIntegerID, CreatedMixin, MappedAsDataclass):
     __tablename__ = "moderation_log"
 
-    moderator_id: Mapped[int] = mapped_column(ForeignKey("user.id", ondelete="CASCADE"))
-    moderator = relationship("User")
-
-    annotation_id: Mapped[types.URLSafeUUID] = mapped_column(
-        ForeignKey("annotation.id", ondelete="CASCADE"), index=True
-    )
-    annotation = relationship("Annotation")
+    moderator: Mapped["User"] = relationship("User")
```

Comment:
Adjusting the dataclass definition to able to instantiate objects just passing annotation (vs `annotation_id`)